### ISSUE TITLE
Limiting users access to the Wazuh app

### ISSUE THREAD

**Issue by rossengeorgiev** [wazuh/wazuh-kibana-app#335]
When X-Pack is enabled there is no way to limit user access to the Wazuh App.
Removing permission to read data from system indices doesn't prevent the a user from browsing around, viewing agents, adding, restarting etc.

**Comment by jesusgn90** [comment]
Hi @rossengeorgiev take a look at this comment, another user just asked it this days, you could read my description about X-Pack and role management, thanks.

https://github.com/wazuh/wazuh-kibana-app/issues/324#issuecomment-375597836

Best regards,
Jesús

**Comment by jesusgn90** [comment]
Hi @rossengeorgiev , and @JPLachance please look at this issue here there is more information about our last messages, here is a little summary from our last steps with X-Pack, hope it helps you guys:

Update X-Pack roles and Wazuh App
=============================================

From my last researchs I concluded the following roles. But first step is to describe
a real scenario where it could be applied:

_We need an user with the minimum privileges to starts the Kibana backend,
which includes the whole Kibana plugins and the Wazuh App internal calls needed
to fetch all the data. This way we can replace the `elastic` user._

_On the other hand we have a user who needs to login through the UI with minimum privileges
to see the visualizations from `wazuh-alerts*` and `wazuh-monitoring*`._

__Note: this feature will be released but just now it won't work on our current packages.__

## User who starts Kibana 

The user who will be written under `kibana.yml` in order to connect with Elasticsearch.

```
...
elasticsearch.username="admin" 
elasticsearch.password="adminadmin"
...
```

This user will has `wazuh-admin` and `kibana_system` roles as you can see here:

![update-1](https://user-images.githubusercontent.com/8860227/37895385-75c8c230-30e1-11e8-8a07-f8d398689e71.png)

The `wazuh-admin` role is as follow:

![update-2](https://user-images.githubusercontent.com/8860227/37895387-75ec98a4-30e1-11e8-8e8c-40d962ebc962.png)

The `kibana_system` role is a built-in role from Kibana ([https://www.elastic.co/guide/en/x-pack/6.2/built-in-roles.html](https://www.elastic.co/guide/en/x-pack/6.2/built-in-roles.html)).

## Standard user who uses the Wazuh App UI

The user who will log in through the UI and will navigate the Wazuh App, but doesn't need to modify the data and/or view certain indices.

As we can see here:

![update-3](https://user-images.githubusercontent.com/8860227/37895389-7616a8a6-30e1-11e8-9ede-bd63518deed3.png)

It will has three roles: `kibana-wazuh-user`, `wazuh-basic-role` and `wazuh-basic-alerts`.

It will need to read the `.kibana` index:

![update-4](https://user-images.githubusercontent.com/8860227/37895390-763d36d8-30e1-11e8-88df-e3d527c7289e.png)

It will also need to read the `.wazuh` and `.wazuh-version` indices:

![update-5](https://user-images.githubusercontent.com/8860227/37895391-765fbaaa-30e1-11e8-87d2-3d44a1f564cd.png)


Finally, it will need to read from `wazuh-alerts-3.x-*` and `wazuh-monitoring-3.x-*`

![update-6](https://user-images.githubusercontent.com/8860227/37895392-768e1c38-30e1-11e8-87f0-808cb316d2dc.png)

This way we can have multiple users sharing `kibana-wazuh-user` and `wazuh-basic-role` roles but not the `wazuh-basic-alerts` role because it could need access different alerts indices.

## What's happening whenever a user can't view anything?

The user will see the following screen on the Wazuh App:

![no-found](https://user-images.githubusercontent.com/8860227/37896085-4c75ceee-30e3-11e8-8947-f4a65c224a46.png)

Best regards,
Jesús

**Comment by rossengeorgiev** [comment]
@jesusgn90 Thanks for the information. What version of the app is that applicable to?

**Comment by jesusgn90** [comment]
Hi @rossengeorgiev currently we have no app for my description, we are working on it on a development branch, but no package is available. It will be released soon but I don't know exactly.

Thanks for your patience, best regards,
Jesús

**Comment by jesusgn90** [comment]
Hello again @rossengeorgiev since this issue will be almost solved with our currently developing branch ([3.2-xpack-rbac](https://github.com/wazuh/wazuh-kibana-app/tree/3.2-xpack-rbac)), I'm going to close this issue. If you have any more trouble, please open a new issue. Stay updated looking the citated branch, or wait for next packages. 

Any case it's probably we publish one or two packages before merge this changes due to it's high unstable nature just now.

Best,
Jesús
